src/main/java/org/opensearch/agent/tools/SearchIndexTool.java

```diff
@@ -77,21 +78,35 @@ public boolean validate(Map<String, String> parameters) {
         return parameters != null && parameters.containsKey(INPUT_FIELD) && parameters.get(INPUT_FIELD) != null;
     }
 
+    private SearchRequest getSearchRequest(String index, String query) throws IOException {
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+        XContentParser queryParser = XContentType.JSON.xContent().createParser(xContentRegistry, LoggingDeprecationHandler.INSTANCE, query);
+        searchSourceBuilder.parseXContent(queryParser);
+        return new SearchRequest().source(searchSourceBuilder).indices(index);
+    }
+
     @Override
     public <T> void run(Map<String, String> parameters, ActionListener<T> listener) {
         try {
             String input = parameters.get(INPUT_FIELD);
             JsonObject jsonObject = StringUtils.gson.fromJson(input, JsonObject.class);
             String index = jsonObject.get(INDEX_FIELD).getAsString();
             String query = jsonObject.get(QUERY_FIELD).toString();
-            query = "{\"query\": " + query + "}";
-
-            SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
-            XContentParser queryParser = XContentType.JSON
-                .xContent()
-                .createParser(xContentRegistry, LoggingDeprecationHandler.INSTANCE, query);
-            searchSourceBuilder.parseXContent(queryParser);
-            SearchRequest searchRequest = new SearchRequest().source(searchSourceBuilder).indices(index);
+
+            SearchRequest searchRequest;
+            try {
+                searchRequest = getSearchRequest(index, query);
+            } catch (Exception e1) {
+                try {
+                    // try different json parsing method
+                    query = jsonObject.get(QUERY_FIELD).getAsString();
```

Comment:
Any test cover this line? Is it safe to invoke `getAsString()` on an jsonElement?

Reply:
There is a test case named `testRunWithEmptyQueryBody` to cover this line. Sometimes the query is `"{}"`. If the previous parsing method fails, we can try parse with `getAsString()`. If there is any exception, this tool will call `listener.onFailure(e);`